### PR TITLE
fix some errors in ConfigFiles.pm

### PR DESCRIPTION
**Pull request by dg4prez**

### CODE REVIEW

**Comment by mitchell852** [comment]
@dg4prez - can you create an issue this so we can understand what the problem was?

**Comment by dg4prez** [comment]
https://github.com/Comcast/traffic_control/issues/1871

**Comment by dg4prez** [comment]
[2016-09-20 16:38:48,287] [DEBUG] GET "/genfiles/view/772/all".
[2016-09-20 16:38:48,289] [DEBUG] Routing to controller "UI::ConfigFiles" and action "genfiles".
DBIx::Class::Storage::DBI::select_single(): Query returned more than one row.  SQL that returns multiple rows is DEPRECATED for ->find and ->single at /home/dgelin200/cdn/traffic_control/traffic_ops/app/lib/UI/ConfigFiles.pm line 1277
[2016-09-20 16:38:51,649] [DEBUG] MID PARENT.CONFIG:
**Removed**

[2016-09-20 16:38:52,069] [DEBUG] Rendering template "config_files/genfiles.html.ep".

File: traffic_ops/app/lib/UI/ConfigFiles.pm [hunk 1]
```diff
@@ -945,7 +945,7 @@ sub remap_dot_config {
 			if ( $remap->{type} =~ /LIVE/ && $remap->{type} !~ /NATNL/ ) {
 				next;    # Live local delivery services skip mids
 			}
-			if ( defined( $mid_remap{ $remap->{org} } ) ) {
+			if ( defined( $mid_remap{ defined( $remap->{org} ) } ) ) {
```
Comment (mitchell852) [inline traffic_ops/app/lib/UI/ConfigFiles.pm]: this doesn't seem right to me but i'm no perl expert. @dangogh ?

Comment (mitchell852) [inline traffic_ops/app/lib/UI/ConfigFiles.pm]: maybe this instead? 

if ( defined( $remap->{org}) && defined( $mid_remap{$remap->{org} ) )

but maybe that's the same result as what you did?

File: traffic_ops/app/lib/UI/ConfigFiles.pm [hunk 2]
```diff
@@ -220,7 +220,7 @@ sub ds_data {
 		my $multi_site_origin           = $row->multi_site_origin;
 		my $multi_site_origin_algorithm = $row->multi_site_origin_algorithm;
 
-		if ( $re_type eq 'HOST_REGEXP' ) {
+		if ( $re_type eq 'HOST_REGEXP' && $ds_type ne 'ANY_MAP' ) {
```
Comment (mitchell852) [inline traffic_ops/app/lib/UI/ConfigFiles.pm]: is this change related to #1871 or was something else broken?

Comment (dg4prez) [inline traffic_ops/app/lib/UI/ConfigFiles.pm]: Just 1871.  Mark and I noticed the errors when we were reviewing the MSO parent.config changes I submitted.

**Comment by mitchell852** [comment]
@dg4prez - if you can get the suggested edits made, I can merge this PR

**Comment by dg4prez** [comment]
Changes made.
